[MI-2458] Added the support for multiple posts for carousel images

manojmalik20 opened the pull request:
** Please ignore the CI failing as I'll fix it later **

- Added the logic of adding as many images in a post as possible and then creating new post
- Added the logic of storing the post ids in KV store
- Added the logic of deleting other posts and updating the selected post on clicking
- Added the logic of removing the post action buttons from posts when no action is taken on those posts

manojmalik20 committed:
[MI-2458] Added the support for multiple posts for carousel images
Added the logic of adding as many images in a post as possible and then creating new post
Added the logic of storing the post ids in KV store
Added the logic of deleting other posts and updating the selected post on clicking
Added the logic of removing the post action buttons from posts when no action is taken on those posts

manojmalik20 committed:
[MI-2458] Modified the appearance of the selected post after an image is selected
Created some new constants and added more values in the context map of post actions

manojmalik20 committed:
[MI-2458] Optimized the approach for showing the image in the selected post of carousel
Removed some constants

ayusht2810 commented on server/plugin/virtual_agent.go:
```diff
@@ -512,31 +514,103 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
+			post, err := p.API.GetPost(postID)
+			if err != nil {
+				p.API.LogDebug("Unable to get the post", "PostID", postID, "Error", err.Error())
+			}
+			if post == nil {
```
```suggestion
			}
			
			if post == nil {
```

ayusht2810 commented on server/plugin/virtual_agent.go:
```diff
@@ -512,31 +514,103 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
+			}
+			if post == nil {
+				return
+			}
+			attachments := post.Attachments()
```
```suggestion
			}
			
			attachments := post.Attachments()
```

manojmalik20 committed:
[MI-2458] Fixed the failing unit tests and lint errors
Moved several structs in the serializer package
Created new mocks and updated the references to structs

manojmalik20 committed:
Merge branch 'epic_phase_3' of github.com:Brightscout/mattermost-plugin-servicenow-virtual-agent into MI-2458

manojmalik20 committed:
Merge branch 'epic_phase_3' of github.com:Brightscout/mattermost-plugin-servicenow-virtual-agent into MI-2458

raghavaggarwal2308 commented on server/plugin/api.go:
```diff
@@ -446,39 +446,83 @@ func (p *Plugin) handlePickerSelection(w http.ResponseWriter, r *http.Request) {
+
+			if err = p.store.StorePostIDs(userID, make([]string, 0)); err != nil {
+				p.API.LogDebug("Unable to store the post IDs in KV store", "UserID", userID, "Error", err.Error())
+			}
```
What are we doing here?

manojmalik20 replied:
We are clearing the IDs that we stored.

manojmalik20 committed:
[MI-2458] Minor changes

Kshitij-Katiyar commented on server/plugin/virtual_agent.go:
```diff
@@ -454,34 +315,108 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
+}
+
+func IsCharCountSafe(attachments []*model.SlackAttachment) bool {
+	bytes, _ := json.Marshal(attachments)
```
we can at least log this error from marshal in case

manojmalik20 committed:
[MI-2458] Review fixes: Logged an error while marshalling

manojmalik20 committed:
[MI-2458] Fixed lint error
